### ISSUE TITLE
Unable to resolve a haste module after upgrading to React Native 0.56

### ISSUE THREAD

**Issue by jsdario** [facebook/metro#200]
Until React Native 0.55 –included– I could call `import {something} from 'hasteModuleSomething'`, where `hasteModuleSomething.js` existed and had the following lines on the top:

```
/**
* @providesModule hasteModuleSomething
* @flow
*/
```

It also breaks the test suite, that uses jest.

**What is the expected behavior?**
I should be able to keep using the current haste modules architecture, at least while the company moves slowly towards a more conventional imports strategy.

**Please provide your exact Metro configuration and mention your Metro, node, yarn/npm version and operating system.**

```
// rn-cli.config.js
const blacklist = require('metro/src/blacklist')

module.exports = {
  getBlacklistRE () {
    return blacklist([/react-native\/local-cli\/core\/__fixtures__.*/])
  },
}
```

Building from a MacOSX, Node.js v8.13, yarn v1.7.0, React v16.4

If this is not a bug (probably is a design decision) could you give me some directions on how to recover the functionality? I opened a SO question in such case, so the correct answer can be rewarded.

https://stackoverflow.com/questions/51378995/unable-to-resolve-a-haste-module-after-upgrading-to-react-native-0-56

Thanks

**Comment by nitrosalat** [comment]
@jsdario did you resolved this issue? I encountered the same issue at the same environment configuration.

**Comment by jsdario** [comment]
I have not, I did change all imports to other structure with ~100 search and replace operations

**Comment by rafeca** [comment]
Thanks for reporting! This has been fixed on RN 0.57.0

**Comment by arled** [comment]
and broken again in RN 0.57.2
